src/reducers/articles.js

```diff
@@ -1,107 +1,87 @@
 }
-
-function tagArticles(state = {}, action) {
-  const response = action.response
-  let rtn = {}
+function articles(state = {}, action) {
+  let key = generateKey(action.tags)
+  let _state = {}
   switch (action.type) {
-    case ActionType.LOADED_MULTI_TAGGED_ARTICLES_SUCCESS:
-      /* data structure:
-          response: {
-            results: {
-              0: {
-                _items: [ ... ]
-              },
-              1 : {
-                _items: [ ... ]
-              }
-            }
-          }
-      */
-      if (response) {
-        let results = response.results
-        let tags = action.tags || []
-        for (let i =0; i < tags.length; i++) {
-          let tag = tags[i]
-          let result = results[i]
-          let items = result.items || []
-          let total = result.meta && result.meta.total
-          rtn[tag] = {
-            items: items,
-            total: total,
-            hasMore: total > items.length ? true : false
-          }
-        }
+    case types.FETCH_ARTICLES_REQUEST:
+      if (state.hasOwnProperty(key)) {
+        merge(_state, state[key], {
+          isFetching: true
+        })
+      } else {
+        merge(_state, {
+          isFetching: true,
+          error: null,
+          nextUrl: null,
+          items: []
+        })
       }
-      return Object.assign({}, state, rtn)
-
-    case ActionType.LOADED_ARTICLES_SUCCESS:
-      /* data structure:
-          response: {
-            results: {
-              _items: [ ... ]
-            }
-          }
-      */
-      if (response) {
-        let tag = ''
-        if ('string' === typeof action.tags) {
-          tag = action.tags
-        } else if (Array.isArray(action.tags)) {
-          tag = action.tags.toString()
-        }
-
-        let items = response.items
-        const meta = response.meta
-
-        items = state[tag] ? state[tag].items.concat(items) : items
-
-        rtn[tag] = {
-          items: items,
-          total: meta.total,
-          hasMore: meta.total > items.length ? true : false
-        }
+      return merge({}, state, {
+        [ key ]: _state
+      })
+    case types.FETCH_ARTICLES_FAILURE:
+      if (state.hasOwnProperty(key)) {
+        merge(_state, state[key], {
+          isFetching: false,
+          error: action.error,
+          lastUpdated: action.failedAt
+        })
+      } else {
+        merge(_state, {
+          isFetching: false,
+          error: action.error,
+          nextUrl: null,
+          items: [],
+          lastUpdated: action.failedAt
+        })
       }
-      return Object.assign({}, state, rtn)
-
-    case ActionType.LOADED_MULTI_TAGGED_ARTICLES_FAILURE:
-    case ActionType.LOADED_ARTICLES_FAILURE:
-      return state
-
+      return merge({}, state, {
+        [ key ]: _state
+      })
+    case types.FETCH_ARTICLES_SUCCESS:
+      let nextUrl = ''
+      let response = action.response
+      try {
+        let href = action.response.links.next.href
```

Comment:
do you want to make the variable name be 'nextHref' or something that people can understand?